### PR TITLE
Add tests to check from_string & from_spec equality

### PR DESCRIPTION
**Pull request by Hritik14**
This will avoid errors like https://github.com/nexB/univers/pull/27
popping up in future

Signed-off-by: Hritik Vijay <hritikxx8@gmail.com>

### CODE REVIEW

**Review by pombredanne — APPROVED** [review]
Thank you : LGTM!

File: tests/test_version_range.py [hunk 1]
```diff
@@ -150,3 +152,19 @@ def test_GemVersionRange_from_native_range_with_pessimistic_operator(self):
             VersionConstraint(comparator=">=", version=RubygemsVersion(string="2.0.8")),
             VersionConstraint(comparator="<", version=RubygemsVersion(string="2.1")),
         )
+
+
+@pytest.mark.parametrize("scheme, range_class", RANGE_CLASS_BY_SCHEMES.items())
+class TestVersionRangeImplementations:
+    VERSION_RANGES_BY_SCHEMES = {
```
Comment (pombredanne) [inline tests/test_version_range.py:159]: What about making this what you parametrize on?
And adding a separate, non-parametrized test that fails if there is not one test for every supported/known vers/purl types?

File: tests/test_version_range.py [hunk 2]
```diff
@@ -150,3 +152,19 @@ def test_GemVersionRange_from_native_range_with_pessimistic_operator(self):
...
+        "nginx": ["0.8.40+", "0.7.52-0.8.39", "0.9.10", "1.5.0+, 1.4.1+"],
+    }
+
+    def test_from_native_and_from_string_equality(self, scheme, range_class):
+        if scheme not in self.VERSION_RANGES_BY_SCHEMES:
```
Comment (pombredanne) [inline tests/test_version_range.py:164]: As mentioned above, you should not iterate and skip if not present. Instead iterate on the actual tests  and not the range class... and have a test that fails if we do not cover all the schemes.

Comment (Hritik14) [reply tests/test_version_range.py:164]: But this test will almost certainly fail as we have many unimplemented VersionRange

Comment (pombredanne) [reply tests/test_version_range.py]: not if marked as expected failure

Comment (Hritik14) [reply tests/test_version_range.py]: Thanks!

**Pull request merged**